### PR TITLE
Us006 expected responsabilities

### PR DESCRIPTION
**Pull request by ruxiiiiii**

### CODE REVIEW

File: frontend/viewdir/jobRolesPage.njk [hunk 1]
```diff
@@ -13,4 +13,5 @@
             </div>
         {% endfor %}
     </div>
+
```
Comment (stephenmcguigan-kainos) [inline frontend/viewdir/jobRolesPage.njk:16]:
```suggestion
```

File: frontend/viewdir/responsabilities.njk [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{% extends "template.njk" %}
```
Comment (stephenmcguigan-kainos) [inline frontend/viewdir/responsabilities.njk:1]: Is this file used or can it be removed?

File: frontend/src/routes.js [hunk 3]
```diff
@@ -3,6 +3,8 @@ const nunjucks = require('nunjucks');
 const path = require('path');
 const jobs = require('./jobRoles.js')
 const caps = require('./capabilities.js')
+
+const api_url = process.env.API_URL;
```
Comment (stephenmcguigan-kainos) [inline frontend/src/routes.js:7]: This was previously removed its only used in the only js files

File: frontend/src/routes.js [hunk 4]
```diff
@@ -3,6 +3,8 @@ const nunjucks = require('nunjucks');
 const path = require('path');
 const jobs = require('./jobRoles.js')
 const caps = require('./capabilities.js')
+
```
Comment (stephenmcguigan-kainos) [inline frontend/src/routes.js:6]:
```suggestion
```

File: backend/JavaWebService/src/main/java/com/kainos/ea/db/RoleMapper.java [hunk 5]
```diff
@@ -7,7 +7,12 @@
 
 @Mapper
 public interface RoleMapper {
-    @Select("select roleID, roleName, roleDesc, datePosted, hours, location, bandID, capabilityID, link from roles where roleID = #{roleID}")
+    @Select("" +
```
Comment (stephenmcguigan-kainos) [inline backend/JavaWebService/src/main/java/com/kainos/ea/db/RoleMapper.java:10]: Why the empty string?

File: backend/JavaWebService/src/main/java/com/kainos/ea/db/RoleMapper.java [hunk 6]
```diff
@@ -7,7 +7,12 @@
...
+            "SELECT r.roleID, r.roleName, r.roleDesc, r.datePosted, r.hours, r.location, r.bandID, r.capabilityId, r.link," +
+            "group_concat(re.responsibility) as jobResponsibilities FROM roles r" +
+            " LEFT JOIN role_responsibilities rr ON r.roleID = rr.roleId" +
+            " LEFT JOIN responsibilities re ON rr.responsibilityId = re.responsibilityId WHERE r.roleID = #{roleID}" +
```
Comment (stephenmcguigan-kainos) [inline backend/JavaWebService/src/main/java/com/kainos/ea/db/RoleMapper.java:14]: Maybe put the where on a new line if everything else is like that?

File: SQLScripts/CreateTables_ATeam.sql [hunk 7]
```diff
@@ -30,6 +30,20 @@ CREATE TABLE `band` (
...
+  `responsibilityId` smallint NOT NULL,
+  `responsibility` text NOT NULL,
+   PRIMARY KEY (`responsibilityId`)
+ );
+
+ CREATE TABLE `role_responsibilities` (
+  `roleId` smallint NOT NULL,
+  `responsibilityId` smallint NOT NULL,
+   PRIMARY KEY (`roleId`,`responsibilityId`),
+   KEY `responsibilityId` (`responsibilityId`),
```
Comment (stephenmcguigan-kainos) [inline SQLScripts/CreateTables_ATeam.sql:43]: Whats key do here?

**Review by stephenmcguigan-kainos — CHANGES_REQUESTED** [review]

File: frontend/viewdir/responsabilities.njk (hunk 2 above)
Comment (AmamMcMam) [reply frontend/viewdir/responsabilities.njk:1]: Also if the page is needed, can it be renamed to "responsibilities.njk"

File: frontend/src/routes.js [hunk 8]
```diff
@@ -3,6 +3,7 @@ const nunjucks = require('nunjucks');
 const path = require('path');
 const jobs = require('./jobRoles.js')
 const caps = require('./capabilities.js')
+const api_url = process.env.API_URL;
```
Comment (stephenmcguigan-kainos) [inline frontend/src/routes.js:6]:
```suggestion
```

**Review by stephenmcguigan-kainos — APPROVED** [review]

**Review by TCprof — APPROVED** [review]

**Pull request merged**